src/ModuleManager.php

```diff
@@ -57,27 +55,49 @@ private static function getPackageSubscriptions(PackageInterface $package): arra
      * @param PackageInterface $package Composer package containing this module.
      * @throws ModuleAlreadyExistsException if the module is already installed.
      * @throws ClassNotFoundException if an event subscription class cannot be resolved.
-     * @throws WrongTypeException if an event subscription class does not implement the EventHandler
+     * @throws WrongTypeException if an event subscription class does not implement the Module
      * interface or the pattern is not a valid regular expression.
      */
     public function register(string $library, PackageInterface $package)
     {
-        $m = $this->g->getEntityManager()->getRepository(Module::class)->find($library);
+        $m = $this->g->getEntityManager()->getRepository(\LotGD\Core\Models\Module::class)->find($library);
         if ($m) {
             throw new ModuleAlreadyExistsException($library);
         } else {
             // TODO: handle error cases here.
-            $m = new Module($library);
+            $m = new \LotGD\Core\Models\Module($library);
             $m->save($this->g->getEntityManager());
 
+            $name = $package->getName();
+
+            $class = $package->getExtra()['class'];
+            if (!isset($package->getExtra()['class']) ||
+                !is_string($class))
+            {
+                throw new KeyNotFoundException("Module {$name} is missing a valid 'class' entry in its extra field.");
+            }
+            try {
+                $klass = new \ReflectionClass($class);
+            } catch (\LogicException $e) {
+                throw new ClassNotFoundException("Could not load class ${class} while registering module {$name}.");
+            } catch (\ReflectionException $e) {
+                throw new ClassNotFoundException("Could not find class ${class} while registering module {$name}.");
+            }
+
+            // Verify that the class is a module class.
+            $interface = Module::class;
+            if (!$klass->implementsInterface($interface)) {
```

Comment:
I would prefer to differentiate between could not find and does not implement. Helpful for debugging if the user encounters an exception.